### PR TITLE
fix(TDI-44176): Snowflake add reuse of connection

### PR DESCRIPTION
**Pull request by OleksiiNimych**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-44176
Snowflake components create 2 connections if not use reference connection

**What is the new behavior?**
Snowflake components reuse existing connection

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/340/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/340/)

File: components/components-snowflake/components-snowflake-runtime/src/main/java/org/talend/components/snowflake/runtime/SnowflakeRuntime.java [hunk 1]
```diff
@@ -78,6 +78,14 @@ public Connection createConnection(RuntimeContainer container)
         }
 
         if (container != null) {
+            conn = (Connection) container.getComponentData(container.getCurrentComponentId(), KEY_CONNECTION);
```
Comment (mbasiuk-talend) [inline components/components-snowflake/components-snowflake-runtime/src/main/java/org/talend/components/snowflake/runtime/SnowflakeRuntime.java:81]: It's a duplicate of code, you can extract it to separate method with componentId as a parameter.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/345/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/345/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/344/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/344/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/346/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/346/)

**Pull request merged**